### PR TITLE
Support torch tensors

### PR DESCRIPTION
**Pull request by toyama0919**
support pytorch tensor.
(work as same numpy.)

example.

```py
from PIL import Image
import torchvision
from imgcat import imgcat

path = 'evaluation/car1.jpg'
img = Image.open(path)
to_tensor = torchvision.transforms.ToTensor()
imgcat(to_tensor(img))
```

### CODE REVIEW

File: imgcat/imgcat.py [hunk 1]
```diff
@@ -105,17 +105,21 @@ def to_content_buf(data):
...
-            return buf.getvalue()
+    elif 'torch' in sys.modules and isinstance(data, sys.modules['torch'].Tensor):
+        # numpy ndarray: convert to png
+        im = data
+        if im.shape[0] == 1:
+            mode = 'L'     # 8-bit pixels, grayscale
+            im = im.mul(255).byte().squeeze().numpy()
+        elif im.shape[0] == 3:
+            mode = None    # RGB/RGBA
+            im = im.mul(255).byte().permute(1, 2, 0).numpy()
```
Comment (wookayin) [inline imgcat/imgcat.py]: One should not assume anything RGB/BGR order and datatype (uint8 or float). But why don't we just do `im.numpy()`?

Comment (toyama0919) [reply imgcat/imgcat.py]: hmm, You are right.
fix use torchvision.

File: imgcat/imgcat.py [hunk 2]
```diff
@@ -105,7 +105,6 @@ def to_content_buf(data):
         else:
             raise ValueError("Expected a 3D ndarray (RGB/RGBA image) or 2D (grayscale image), "
                              "but given shape: {}".format(im.shape))
-
```
Comment (wookayin) [inline imgcat/imgcat.py]: nit

File: imgcat/imgcat.py [hunk 3]
```diff
@@ -117,6 +116,20 @@ def to_content_buf(data):
...
             return buf.getvalue()
 
+    elif 'torch' in sys.modules and isinstance(data, sys.modules['torch'].Tensor):
+        # pytorch tensor: convert to png
+        im = data
+        try:
+            from torchvision import transforms
+        except ImportError as e:
+            raise ImportError(e.msg +
+                              "\nTo draw torch tensor, we require torchvision. " +
```
Comment (wookayin) [inline imgcat/imgcat.py]: Could you fix grammar issues here?

**Comment by toyama0919** [comment]
hmm.. python3.4 is not support.
Do you have a good idea?

image

File: imgcat/imgcat.py (hunk 3 above)
Comment (toyama0919) [reply imgcat/imgcat.py]: what is grammar issues?

File: setup.py [hunk 4]
```diff
@@ -39,6 +41,10 @@ def read_version():
 else:  # <= Python 3.4
     tests_requires += ['matplotlib<3.0', 'Pillow<6.0']
 
+# pytorch: python 2.7 require future
+if sys.version_info < (3, 0):
+    tests_requires += ['future']
```
Comment (wookayin) [inline setup.py]: Could you please elaborate on why future is needed?

Comment (toyama0919) [reply setup.py]: python2.7 import error.
` No module named builtins`
for avoid this, Install future.

see.
https://discuss.pytorch.org/t/building-pytorch-from-source-in-conda-fails-in-pytorch-caffe2-operators-crash-op-cc/42859/3
https://github.com/hyperopt/hyperopt/issues/273

```bash
root@1f2b2ea83553:~# pip install torch
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting torch
  Downloading torch-1.4.0-cp27-cp27mu-manylinux1_x86_64.whl (753.4 MB)
     |████████████████████████████████| 753.4 MB 2.5 kB/s
Installing collected packages: torch
Successfully installed torch-1.4.0
root@1f2b2ea83553:~# python
Python 2.7.17 (default, Feb 26 2020, 17:18:08)
[GCC 8.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/torch/__init__.py", line 19, in <module>
    from ._six import string_classes as _string_classes
  File "/usr/local/lib/python2.7/site-packages/torch/_six.py", line 23, in <module>
    import builtins
ImportError: No module named builtins
```

**Comment by wookayin** [comment]
This works for me (numpy):
```
❯❯❯ python -c 'import imgcat; from skimage.data import chelsea; imgcat.imgcat(chelsea())'
```

But this does not (torch), not showing any images:
```
❯❯❯ python -c 'import imgcat, torch; from skimage.data import chelsea; imgcat.imgcat(torch.tensor(chelsea()))'
```

Any idea?

**Comment by wookayin** [comment]
Well, I think torchvision has a ~**critical** bug.~ Its data spec is CHW, not HWC. I wonder you did not even test it?

```
>>> a = torch.tensor(skimage.data.chelsea())
>>> a.shape
torch.Size([300, 451, 3])

>>> transforms.ToPILImage()(a)
<PIL.Image.Image image mode=RGB size=3x451 at 0x12F0C57D0>
```

ToPILImage():
```
Converts a torch.*Tensor of shape C x H x W or a numpy ndarray of shape
    H x W x C to a PIL Image while preserving the value range.
```

**Comment by wookayin** [comment]
Sorry, I now realize that torch always use CHW convention, so it's not imgcat's fault and we may not need to convert them.

```python
from PIL import Image
import torchvision.transforms.functional as TF
image = Image.open('screenshot.png')
x = TF.to_tensor(image)    # [3, H, W]
imgcat(x0
```

and it works.

**Comment by toyama0919** [comment]
https://github.com/wookayin/python-imgcat/pull/7#issuecomment-596033185
What about this problem?
(Though the part of the torch can skip 3.4 etc..

**Comment by toyama0919** [comment]
Please also release to pypi 🙏

**Comment by wookayin** [comment]
@toyama0919 Released v0.5.0 to pypi.

**Pull request merged**